### DISCUSSION TITLE
Fill and Empty Shovel Mode

### DISCUSSION THREAD

**Discussion by chrisagreat** [Courseplay/courseplay#7651] [Q&A]
2 part fill and empty shovel question:::::::

1 ------------------

I'm using the fill and empty shovel mode.... 98% of it works.... except when I go to dump into a trailer on autodrive.... The tractor sees 'traffic' and goes wacky to avoid it, or stops... (debug is on, you can see the collision detection on the ground) I've turned off collision detection in auto drive settings, but no good.... are there other places to turn off and all detection for this..... I almost have a full bunker silo to ferment and sell....

Also all 4 shovel positions are set correct and it sees and fills from the bunker, and drives to the trailer but it just 'traffic' stops too soon and can't dump.

How can I disable/change this so the loader works properly with a trailer.

![20211209031144_1](https://user-images.githubusercontent.com/95962290/145690823-b1c91427-ad3c-4155-b3a5-f8e1339920b7.jpg)

2 ------------------

I also tried adding a silo to the area, and made a course to run over the silo, drop, and move on, but that didn't work either. The loader just goes in circles, but it does see and empty the bunker. I have tried a wait, an unload, and both. the loader doesn't stop.

How do I stop the loader on the silo?

![20211210192759_1](https://user-images.githubusercontent.com/95962290/145690704-287494b2-af65-460a-8d4b-36add01dd9ac.jpg)

I'm using this model to set up the fill and empty.

![Courseplay - Fill + Empty Shovel](https://user-images.githubusercontent.com/95962290/145690738-b73b2fca-3e6f-4819-b6d2-7ad3a516ddc9.jpg)

Thanks for your help!!!
Chris

**Comment by Tensuko** [comment]
What Version do you use ?
The Pattern is for old Versions, thats why I ask.

You may want to try the latest Version from ModHub and use the New Course Pattern. But I am not sure if the collision detection is still a problem when unloading into a trailer. If CP stays there to load from Trailer, it is working.
